File: org.tec.datastructures/src/org/tec/datastructures/structures/Stack.java
package org.tec.datastructures.structures;

import org.tec.datastructures.nodes.SimpleNode;

public class Stack<T extends Comparable<T>>{
	
	private SimpleNode<T> head;
	
	public Stack() {
		this(null);
	}
	
	public Stack(SimpleNode<T> Head) {
		head = Head;
	}
	
	public SimpleNode<T> peak(){
		return this.head;
	}
	
	
	public void push (T value) {
		if (head == null){
			head = new SimpleNode<T>(value);
			return;
		}
		SimpleNode<T> appended = new SimpleNode<T>(value);
		appended.setNext(head);
		head = appended;
	}

	
	public SimpleNode<T> pop() {
		if (head == null) {
			return null;
		}
		SimpleNode<T> popped = head;
		head = head.getNext();
		return popped;
	}

	public int size() {
		SimpleNode<T> current = head;
		int result = 0;
		while (current != null){
			result++;
			current = current.getNext();
		}
		return result;
	}

	public void clear() {
		this.head = null;
	}
	
}

File: C#/org.tec.datastructures/Stack.cs
using System;
namespace org.tec.datastructures
{
    public class Stack<T>
    {
        SimpleList<T> container = new SimpleList<T>();

        public void Push(T value){
            container.Append(value);
        }

        public T Peek(){
            return container.Get(container.Length() - 1);
        }

        public T Pop(){
            T value = container.Get(container.Length() - 1);
            container.Delete(container.Length() - 1);
            return value;
        }

        public Boolean IsEmpty(){
            return container.IsEmpty();
        }
    }
}
